File: DemoMethods/Basic/Edit.cs
using System;
using System.Web.Http;
using DemoMethods.Entities;
using DemoMethods.Helpers;

namespace DemoMethods.Basic
{
    public partial class BasicController : DemoApiController
    {
      
        [HttpGet]
        public object Edit(int c)
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                // Edit :
                var company = session.Load<Company>(c);
                company.Address.Line2 = "Zip 12345";
             
                session.SaveChanges();

                return (company.Id);
            }
        }

    }
}

File: DemoMethods/PythonDemo/demo_methods/Basic/Edit.py
from demo.holder import DocumentStoreHolder
from demo.entities import *
from pyravendb.data.document_convention import DocumentConvention


class Edit(object):
    @staticmethod
    def edit(company_id=1):
        with DocumentStoreHolder.get_store().open_session() as session:
            company_id = "{0}/{1}".format(DocumentConvention.default_transform_type_tag_name(Company.__name__), company_id)

            company = session.load(company_id, object_type=Company,
                                   nested_object_types={"Address": AddressC, "Contact": ContactC})

            company.Address.Line2 = "Zip 12345"
            session.save_changes()
            
            return "CompanyId {0} Edited Successfully.".format(company_id)
